code/game/objects/structures/crates_lockers/closets.dm

```diff
@@ -457,3 +464,17 @@
 
 /obj/structure/closet/return_temperature()
 	return
+
+/obj/structure/closet/proc/dive_into(mob/living/user)
+	if(!open() && !opened)
+		togglelock(user, TRUE)
+		if(!open())
+			to_chat(user, "<span class='warning'>It won't budge!</span>")
+			return
+	user.visible_message("<span class='warning'>[user] dives into [src] and slams the door shut behind them!</span>", \
+	"<span class='danger'>You dive into [src]!</span>")
+	user.forceMove(src)
+	user.resting = TRUE //so people can jump into crates without slamming the lid on their head
+	close()
+	user.resting = FALSE

```

Comment:
I have a feeling you should check if it's a crate before resting the user otherwise even though this is consistent behaviour, it isn't consistent with hiding in a locker.

Is it intended balance to not allow someone to hold a stunbaton or revolver in their hand as they do this in a panic?

Reply:
Am I just ignorant and don't know how this works out in practice?

Reply:
this won't cause people to drop items, since it sets the var directly

Reply:
Yup, it doesn't cause any item dropping.

Reply:
You should probably check if people are dragging the closet/crate, as I can totally see people just dragging them around and diving into them once people start shooting. Maybe add a slight stun so jumping in to get out-of-sight of a person is still very viable but jumping in to avoid gunfire is probably a lot less effective since they can just open the locker and shoot you while you're stunned.